contracts/GovHub.sol

```diff
@@ -195,21 +233,18 @@ contract GovHub is System, IParamSubscriber {
 
   /// Cancel the proposal, can only be done by the proposer
   /// @param proposalId The proposal Id
-  function cancel(uint256 proposalId) public onlyInit {
+  function cancel(uint256 proposalId) public onlyInit onlyIfProposer(proposalId){
     ProposalState state = getState(proposalId);
     require(state == ProposalState.Pending || state == ProposalState.Active, "cannot cancel finished proposal");
 
     Proposal storage proposal = proposals[proposalId];
-    require(msg.sender == proposal.proposer, "only cancel by proposer");
-
     proposal.canceled = true;
     emit ProposalCanceled(proposalId);
   }
 
   /// Execute the proposal
```

Comment:
So I would make it into a tech-debt rather than delay this PR. I also believe the functions we have documented here are the major in terms of describing our overall process

Reply:
I'm okay with that